docs/docs/writing-plugins/macros.mdx

```diff
@@ -107,3 +107,39 @@ custom_python_sources(
     has_type_hints=False,
 )
 ```
+
+## Documenting your macros
+
+Using doc-strings to document your macros, Pants will pick up and present these for the online help text. Also global constants can be documented using `Annotated`.
+
+```python title="pants-plugins/macros.py"
+OUR_GLOBAL_CONSTANT: Annotated[
+  int,
+  """This is our magic number.
```

Comment:
This is a bit of a bikeshed discussion, but my understanding is that strings within type annotations need to be carefully considered, due to deferred annotations (where the annotation (or parts of it) ends up being stored as a string and lazily `eval`'d when required). An alternative might be a dedicated marker class, e.g.

```python
class Doc:
    def __init__(self, s): ...
```

Used like:


```suggestion
OUR_GLOBAL_CONSTANT: Annotated[
  int,
  Doc(
      """This is our magic number.
```

(This is directly inspired by https://peps.python.org/pep-0727/, and we could potentially even use `typing_extensions.Doc` coming from that pep)

Reply:
The string part is opaque to the type system, it is up to tooling/programs encountering it to treat it.. and as such ought to be resilient against unexpected types as it's arbitrary metadata for anyone to use.

https://docs.python.org/3/library/typing.html#typing.Annotated
> If a library or tool encounters an annotation Annotated[T, x] and has no special logic for the metadata, it should ignore the metadata and simply treat the annotation as T. As such, Annotated can be useful for code that wants to use annotations for purposes outside Python’s static typing system.
>
> Using Annotated[T, x] as an annotation still allows for static typechecking of T, as type checkers will simply ignore the metadata x. In this way, Annotated differs from the [@no_type_check](https://docs.python.org/3/library/typing.html#typing.no_type_check) decorator, which can also be used for adding annotations outside the scope of the typing system, but completely disables typechecking for a function or class.
>
> The responsibility of how to interpret the metadata lies with the tool or library encountering an Annotated annotation. A tool or library encountering an Annotated type can scan through the metadata elements to determine if they are of interest (e.g., using [isinstance()](https://docs.python.org/3/library/functions.html#isinstance)).

Reply:
Ah, in that direction. Gotcha. Unlikely, but I don't disagree.. I'll have a looksie ;)